### DISCUSSION TITLE
ezEngine now fully migrated to Jolt Physics

### DISCUSSION THREAD

**Discussion by jankrassnigg** [jrouwe/JoltPhysics#292] [Show and tell]
In mid February we set out to mirgrate [our engine](https://github.com/ezEngine/ezEngine) to Jolt.

I am happy to announce, that this migration has now reached feature parity with what we had before and thus we have now fully switched to Jolt and will continue all physics feature development exclusively with Jolt.

Working with Jolt has been (and generally is) very enjoyable. It is really well designed, easy to understand and performance and stability are great. Sure, every engine has to make trade-offs here and there, but Jolt is really designed for games scenarios, and in my opinion makes exactly the right decisions where to make tradeoffs.

Here is a list of all the things we have now implemented in EZ:

* Static, dynamic and kinematic bodies
* Box, sphere, capsule, cylinder shapes
* Triangle collision meshes with different materials
* Per material friction and restitution
* Convex collision shapes
* Convex decomposition of shapes (using [V-HACD](https://github.com/kmammou/v-hacd))
* Per body material overrides
* Configurable collision layers
* Custom center of mass option
* Jolt debug visualization of bodies and constraints
* Trigger volumes
* Constraints:
  * Point
  * Hinge
  * Slider
  * Cone
  * Swing-Twist
  * Distance
  * Fixed
* Option for bodies to ignore collisions when they are joined with a constraint
* Rope simulation
* Basic Ragdoll simulation (was WIP before Jolt, will be ongoing work in the future)
* Animated mesh hitbox support
* Simulation configuration (fixed stepping, variable stepping and so on)
* Jolt memory tracking
* Base class for using Jolt's virtual character controller
* Additional default implementation of a character controller
* Grab object functionality (to pick up and carry items)
* Raycasts and shape queries and sweep tests with various filter options (by collision layers, general shape types, etc)

![grafik](https://user-images.githubusercontent.com/6001174/197331701-ded93935-7a6c-448d-abb7-303fbb17168b.png)

Big thanks to @jrouwe for this really awesome project and the constant support, It has been a real pleasure and I'm looking forward to tackle more physics features in the future (who knows, maybe one day I'll even look into doing vehicles).

If you want to know more about our engine, have a look at https://ezEngine.net and checkout our [GitHub repo](https://github.com/ezEngine/ezEngine),

**Comment by jrouwe** [comment]
Thank you for your feedback too @jankrassnigg, it has been nice working together!
